File: game.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
游戏逻辑类
"""

import time
import random
from typing import Optional, List
from card import Card, Team, Ability
from player import Player

class Game:
    """游戏主类"""
    def __init__(self):
        self.plant_player: Optional[Player] = None
        self.zombie_player: Optional[Player] = None
        self.current_turn = Team.PLANT
        self.turn_count = 1
        self.game_over = False
        self.winner: Optional[Team] = None
    
    def start_game(self):
        """开始游戏"""
        # 创建玩家
        self.plant_player = Player("植物玩家", Team.PLANT)
        self.zombie_player = Player("僵尸玩家", Team.ZOMBIE)
        
        # 抽取初始手牌
        self.plant_player.draw_initial_hand()
        self.zombie_player.draw_initial_hand()
        
        print("\n游戏开始！")
        print(f"{self.plant_player.name} VS {self.zombie_player.name}")
        
        # 游戏主循环
        while not self.game_over:
            self.play_turn()
            
            # 检查游戏是否结束
            if self.check_game_over():
                break
            
            # 切换回合
            self.switch_turn()
            self.turn_count += 1
        
        self.display_game_result()
    
    def play_turn(self):
        """进行一个回合"""
        current_player = self.get_current_player()
        opponent = self.get_opponent()
        
        print(f"\n{'='*50}")
        print(f"第 {self.turn_count} 回合 - {current_player.team.value}的回合")
        print(f"{'='*50}")
        
        # 开始回合
        current_player.start_turn()
        
        # 显示游戏状态
        self.display_game_state()
        
        # 玩家行动阶段
        self.player_action_phase(current_player)
        
        # 战斗阶段
        self.battle_phase()
        
        # 回合结束阶段
        self.end_turn_phase()
    
    def player_action_phase(self, player: Player):
        """玩家行动阶段"""
        while True:
            print(f"\n{player.name}的行动阶段")
            print("1. 查看手牌")
            print("2. 打出卡牌")
            print("3. 移动卡牌 (仅限有移动能力的卡牌)")
            print("4. 结束回合")
            
            choice = input("请选择操作 (1-4): ").strip()
            
            if choice == "1":
                print(player.get_hand_display())
            
            elif choice == "2":
                self.play_card_action(player)
            
            elif choice == "3":
                self.move_card_action(player)
            
            elif choice == "4":
                print(f"{player.name}结束回合")
                break
            
            else:
                print("无效选择，请重新输入")
    
    def play_card_action(self, player: Player):
        """打出卡牌操作"""
        if not player.hand:
            print("手牌为空！")
            return
        
        print(player.get_hand_display())
        print(player.get_field_display())
        
        try:
            card_index = int(input("选择要打出的卡牌编号 (1-{}): ".format(len(player.hand)))) - 1
            position = int(input("选择放置位置 (1-5): ")) - 1
            
            if player.play_card(card_index, position):
                card = player.field[position]
                print(f"成功打出 {card.name} 到位置 {position + 1}!")
                time.sleep(1)
            else:
                print("无法打出该卡牌，请检查费用或位置!")
        
        except ValueError:
            print("输入无效，请输入数字")
    
    def battle_phase(self):
        """战斗阶段"""
        print("\n⚔️ 战斗阶段开始!")
        time.sleep(1)
        
        # 植物方攻击
        self.team_attack(self.plant_player, self.zombie_player)
        
        # 僵尸方攻击
        self.team_attack(self.zombie_player, self.plant_player)
        
        # 移除死亡卡牌
        self.plant_player.remove_dead_cards()
        self.zombie_player.remove_dead_cards()
        
        print("战斗阶段结束!")
        time.sleep(1)
    
    def move_card_action(self, player: Player):
        """移动卡牌操作"""
        print(player.get_field_display())
        
        try:
            from_pos = int(input("选择要移动的卡牌位置 (1-5): ")) - 1
            to_pos = int(input("选择目标位置 (1-5): ")) - 1
            
            if player.move_card(from_pos, to_pos):
                card = player.field[to_pos]
                print(f"成功将 {card.name} 从位置 {from_pos + 1} 移动到位置 {to_pos + 1}!")
                time.sleep(1)
            else:
                print("无法移动该卡牌，请检查位置或卡牌是否有移动能力!")
        
        except ValueError:
            print("输入无效，请输入数字")
    
    def team_attack(self, attacker: Player, defender: Player):
        """队伍攻击"""
        attacker_cards = [card for card in attacker.get_alive_cards() if card.can_attack()]
        
        if not attacker_cards:
            print(f"{attacker.team.value}没有可攻击的卡牌")
            return
        
        print(f"\n{attacker.team.value}方攻击:")
        
        for card in attacker_cards:
            # 寻找对面的目标
            target = self.find_target(card, defender)
            
            if target:
                killed = self.perform_attack(card, target)
                
                # 疯狂能力：如果击杀了目标，可以再次攻击
                if killed and card.ability == Ability.FRENZY:
                    print(f"{card.name} 发动疯狂效果，可以再次攻击!")
                    new_target = self.find_target(card, defender)
                    if new_target:
                        self.perform_attack(card, new_target)
            else:
                # 没有目标，直接攻击玩家
                damage = card.attack
                defender.take_damage(damage)
                print(f"{card.name} 直接攻击 {defender.name}，造成 {damage} 点伤害!")
            
            # 标记已攻击
            card.has_attacked_this_turn = True
    
    def find_target(self, attacker: Card, defender: Player) -> Optional[Card]:
        """寻找攻击目标"""
        # 优先攻击同一位置的敌人
        same_position = defender.field[attacker.position]
        if same_position and same_position.is_alive():
            return same_position
        
        # 如果同一位置没有敌人，随机选择一个存活的敌人
        alive_enemies = defender.get_alive_cards()
        if alive_enemies:
            return random.choice(alive_enemies)
        
        return None
    
    def perform_attack(self, attacker: Card, defender: Card) -> bool:
        """执行攻击，返回是否击杀目标"""
        damage = attacker.attack
        
        # 处理特殊能力
        attack_count = 1
        ignore_shield = False
        instant_kill = False
        
        if attacker.ability == Ability.DOUBLE_HIT:
            attack_count = 2
            print(f"{attacker.name} 发动双击!")
        
        # 执行多次攻击
        for i in range(attack_count):
            current_damage = damage
            
            # 必中能力：无视防御
            if attacker.ability == Ability.PIERCING:
                ignore_shield = True
                print(f"{attacker.name} 的攻击必中，无视防御!")
            
            # 致命能力：对无防御卡牌一击必杀
            if attacker.ability == Ability.LETHAL and defender.ability != Ability.SHIELD:
                current_damage = defender.current_health
                print(f"{attacker.name} 发动致命一击!")
            
            # 秒杀能力：无视一切防御
            elif attacker.ability == Ability.INSTANT_KILL:
                current_damage = defender.current_health
                ignore_shield = True
                print(f"{attacker.name} 发动秒杀效果!")
            
            print(f"{attacker.name} 攻击 {defender.name}，造成 {current_damage} 点伤害!")
            
            # 造成伤害
            is_dead = defender.take_damage(current_damage, ignore_shield)
            
            if is_dead:
                print(f"{defender.name} 被击败了!")
                
                # 弹回手牌效果
                if attacker.ability == Ability.BOUNCE:
                    defender_player = self.zombie_player if defender.team == Team.ZOMBIE else self.plant_player
                    for i, card in enumerate(defender_player.field):
                        if card == defender:
                            defender_player.bounce_card_to_hand(i)
                            print(f"{defender.name} 被弹回手牌!")
                            break
                
                time.sleep(0.5)
                return True
            else:
                print(f"{defender.name} 剩余生命: {defender.current_health}")
            
            # 冻结效果
            if attacker.ability == Ability.FREEZE:
                defender.freeze()
                print(f"{defender.name} 被冻结了!")
            
            time.sleep(0.5)
        
        return False
    
    def end_turn_phase(self):
        """回合结束阶段"""
        # 治疗能力处理
        self.process_healing_ability()
        
        # 显示当前状态
        self.display_game_state()
    
    def process_healing_ability(self):
        """处理治疗能力"""
        for player in [self.plant_player, self.zombie_player]:
            for card in player.field:
                if card and card.is_alive() and card.ability == Ability.HEAL:
                    # 治疗相邻的卡牌
                    for offset in [-1, 1]:
                        adjacent_pos = card.position + offset
                        if 0 <= adjacent_pos < 5:
                            adjacent_card = player.field[adjacent_pos]
                            if adjacent_card and adjacent_card.is_alive():
                                adjacent_card.heal(1)
                                print(f"{card.name} 治疗 {adjacent_card.name} 1点生命!")
    
    def check_game_over(self) -> bool:
        """检查游戏是否结束"""
        # 检查玩家生命值
        if not self.plant_player.is_alive():
            self.game_over = True
            self.winner = Team.ZOMBIE
            return True
        
        if not self.zombie_player.is_alive():
            self.game_over = True
            self.winner = Team.PLANT
            return True
        
        # 检查是否有一方没有卡牌且手牌为空
        plant_alive = len(self.plant_player.get_alive_cards()) > 0 or len(self.plant_player.hand) > 0
        zombie_alive = len(self.zombie_player.get_alive_cards()) > 0 or len(self.zombie_player.hand) > 0
        
        if not plant_alive:
            self.game_over = True
            self.winner = Team.ZOMBIE
            return True
        
        if not zombie_alive:
            self.game_over = True
            self.winner = Team.PLANT
            return True
        
        return False
    
    def switch_turn(self):
        """切换回合"""
        self.current_turn = Team.ZOMBIE if self.current_turn == Team.PLANT else Team.PLANT
    
    def get_current_player(self) -> Player:
        """获取当前玩家"""
        return self.plant_player if self.current_turn == Team.PLANT else self.zombie_player
    
    def get_opponent(self) -> Player:
        """获取对手"""
        return self.zombie_player if self.current_turn == Team.PLANT else self.plant_player
    
    def display_game_state(self):
        """显示游戏状态"""
        print(f"\n{self.plant_player.get_status_display()}")
        print(self.plant_player.get_field_display())
        
        print(f"\n{self.zombie_player.get_status_display()}")
        print(self.zombie_player.get_field_display())
    
    def display_game_result(self):
        """显示游戏结果"""
        print(f"\n{'='*50}")
        print("游戏结束!")
        print(f"{'='*50}")
        
        if self.winner == Team.PLANT:
            print("🌻 植物队获胜! 🌻")
        else:
            print("🧟 僵尸队获胜! 🧟")
        
        print(f"总回合数: {self.turn_count}")
        print(f"植物玩家剩余生命: {self.plant_player.health}")
        print(f"僵尸玩家剩余生命: {self.zombie_player.health}")

File: game.js
// 游戏核心逻辑
class GameState {
    constructor() {
        this.currentPlayer = null;
        this.opponent = null;
        this.currentTurn = 'plant';
        this.turnCount = 1;
        this.gameActive = false;
        this.selectedCard = null;
        this.selectedTarget = null;
    }
    
    initGame() {
        this.currentPlayer = new Player('植物玩家', 'plant');
        this.opponent = new Player('僵尸玩家', 'zombie');
        this.currentPlayer.initDeck();
        this.opponent.initDeck();
        this.currentPlayer.drawInitialHand();
        this.opponent.drawInitialHand();
        this.gameActive = true;
        this.currentTurn = 'plant';
        this.turnCount = 1;
    }
    
    startTurn() {
        const player = this.getCurrentPlayer();
        player.startTurn();
        this.updateUI();
        this.addBattleLog(`${player.team === 'plant' ? '植物队' : '僵尸队'} 回合 ${this.turnCount}`);
    }
    
    endTurn() {
        this.executeBattlePhase();
        this.switchTurn();
        this.turnCount++;
        this.startTurn();
    }
    
    getCurrentPlayer() {
        return this.currentTurn === 'plant' ? this.currentPlayer : this.opponent;
    }
    
    getOpponent() {
        return this.currentTurn === 'plant' ? this.opponent : this.currentPlayer;
    }
    
    switchTurn() {
        this.currentTurn = this.currentTurn === 'plant' ? 'zombie' : 'plant';
    }
    
    executeBattlePhase() {
        const attacker = this.getCurrentPlayer();
        const defender = this.getOpponent();
        
        // 执行攻击逻辑
        const attackerCards = attacker.getAliveCards();
        attackerCards.forEach(card => {
            if (card.canAttack()) {
                const target = this.findTarget(card, defender);
                if (target) {
                    this.performAttack(card, target, defender);
                } else {
                    // 直接攻击玩家
                    defender.takeDamage(card.attack);
                    this.addBattleLog(`${card.name} 直接攻击 ${defender.name}，造成 ${card.attack} 点伤害`);
                }
            }
        });
        
        // 移除死亡卡牌
        attacker.removeDeadCards();
        defender.removeDeadCards();
        
        // 检查游戏结束
        if (this.checkGameOver()) {
            this.endGame();
        }
    }
    
    findTarget(attacker, defender) {
        // 优先攻击同一位置的敌人
        const samePosition = defender.field[attacker.position];
        if (samePosition && samePosition.isAlive()) {
            return samePosition;
        }
        
        // 随机选择存活的敌人
        const aliveEnemies = defender.getAliveCards();
        if (aliveEnemies.length > 0) {
            return aliveEnemies[Math.floor(Math.random() * aliveEnemies.length)];
        }
        
        return null;
    }
    
    performAttack(attacker, defender, defenderPlayer) {
        let damage = attacker.attack;
        let ignoreShield = false;
        
        // 处理特殊能力
        if (attacker.ability === 'piercing') {
            ignoreShield = true;
            this.addBattleLog(`${attacker.name} 的攻击必中，无视防御`);
        }
        
        if (attacker.ability === 'lethal' && defender.ability !== 'shield') {
            damage = defender.currentHealth;
            this.addBattleLog(`${attacker.name} 发动致命一击`);
        }
        
        if (attacker.ability === 'instantkill') {
            damage = defender.currentHealth;
            ignoreShield = true;
            this.addBattleLog(`${attacker.name} 发动秒杀效果`);
        }
        
        // 造成伤害
        const isDead = defender.takeDamage(damage, ignoreShield);
        
        if (isDead) {
            this.addBattleLog(`${defender.name} 被击败了`);
            
            // 处理弹回效果
            if (attacker.ability === 'bounce') {
                defenderPlayer.bounceCardToHand(defender);
                this.addBattleLog(`${defender.name} 被弹回手牌`);
            }
            
            // 处理疯狂效果
            if (attacker.ability === 'frenzy') {
                this.addBattleLog(`${attacker.name} 发动疯狂效果，可以再次攻击`);
                // 可以再次攻击的逻辑
            }
        } else {
            this.addBattleLog(`${defender.name} 剩余生命: ${defender.currentHealth}`);
        }
        
        // 处理冻结效果
        if (attacker.ability === 'freeze') {
            defender.freeze();
            this.addBattleLog(`${defender.name} 被冻结了`);
        }
        
        attacker.hasAttackedThisTurn = true;
    }
    
    checkGameOver() {
        if (!this.currentPlayer.isAlive() || !this.opponent.isAlive()) {
            return true;
        }
        
        const playerAlive = this.currentPlayer.getAliveCards().length > 0 || this.currentPlayer.hand.length > 0;
        const opponentAlive = this.opponent.getAliveCards().length > 0 || this.opponent.hand.length > 0;
        
        return !playerAlive || !opponentAlive;
    }
    
    endGame() {
        this.gameActive = false;
        const winner = this.currentPlayer.isAlive() ? this.currentPlayer : this.opponent;
        this.addBattleLog(`游戏结束！${winner.name} 获胜！`);
        
        setTimeout(() => {
            alert(`游戏结束！${winner.name} 获胜！`);
            showScreen('mainMenu');
        }, 1000);
    }
    
    addBattleLog(message) {
        const battleLog = document.getElementById('battleLog');
        const logEntry = document.createElement('div');
        logEntry.textContent = message;
        battleLog.appendChild(logEntry);
        battleLog.scrollTop = battleLog.scrollHeight;
    }
    
    updateUI() {
        // 更新玩家信息
        document.getElementById('playerHealth').textContent = this.currentPlayer.health;
        document.getElementById('playerMana').textContent = `${this.currentPlayer.mana}/${this.currentPlayer.maxMana}`;
        document.getElementById('opponentHealth').textContent = this.opponent.health;
        document.getElementById('opponentMana').textContent = `${this.opponent.mana}/${this.opponent.maxMana}`;
        
        // 更新回合信息
        document.getElementById('turnInfo').textContent = `回合 ${this.turnCount} - ${this.currentTurn === 'plant' ? '植物队' : '僵尸队'}`;
        
        // 更新战场
        this.updateBattlefield();
        this.updateHand();
    }
    
    updateBattlefield() {
        const playerField = document.getElementById('playerField');
        const opponentField = document.getElementById('opponentField');
        
        playerField.innerHTML = '';
        opponentField.innerHTML = '';
        
        // 显示玩家战场
        for (let i = 0; i < 5; i++) {
            const slot = document.createElement('div');
            slot.className = 'battlefield-slot';
            slot.dataset.position = i;
            
            const card = this.currentPlayer.field[i];
            if (card && card.isAlive()) {
                slot.appendChild(createCardElement(card));
                
                // 如果有组队联手，显示特殊标记
                if (card.ability === 'teamwork') {
                    const teamworkIcon = document.createElement('div');
                    teamworkIcon.className = 'teamwork-icon';
                    teamworkIcon.innerHTML = '🤝';
                    teamworkIcon.title = '组队联手：可放置第二个植物';
                    slot.appendChild(teamworkIcon);
                }
            }
            
            playerField.appendChild(slot);
        }
        
        // 显示对手战场
        for (let i = 0; i < 5; i++) {
            const slot = document.createElement('div');
            slot.className = 'battlefield-slot';
            slot.dataset.position = i;
            
            const card = this.opponent.field[i];
            if (card && card.isAlive()) {
                slot.appendChild(createCardElement(card));
                
                // 如果有组队联手，显示特殊标记
                if (card.ability === 'teamwork') {
                    const teamworkIcon = document.createElement('div');
                    teamworkIcon.className = 'teamwork-icon';
                    teamworkIcon.innerHTML = '🤝';
                    teamworkIcon.title = '组队联手：可放置第二个植物';
                    slot.appendChild(teamworkIcon);
                }
            }
            
            opponentField.appendChild(slot);
        }
    }
    
    updateHand() {
        const playerHand = document.getElementById('playerHand');
        playerHand.innerHTML = '';
        
        this.currentPlayer.hand.forEach((card, index) => {
            const cardElement = createCardElement(card);
            cardElement.dataset.handIndex = index;
            cardElement.addEventListener('click', () => this.selectCard(card, index));
            playerHand.appendChild(cardElement);
        });
    }
    
    selectCard(card, index) {
        if (this.currentTurn !== 'plant') return; // 只允许玩家操作
        
        this.selectedCard = { card, index };
        
        // 高亮选中的卡牌
        document.querySelectorAll('#playerHand .card').forEach(el => el.classList.remove('selected'));
        document.querySelector(`[data-hand-index="${index}"]`).classList.add('selected');
        
        // 显示可放置的位置
        this.showValidPositions(card);
    }
    
    showValidPositions(card) {
        if (this.currentPlayer.mana < card.cost) return;
        
        const slots = document.querySelectorAll('#playerField .battlefield-slot');
        slots.forEach((slot, index) => {
            const existingCard = this.currentPlayer.field[index];
            
            // 检查是否可以放置
            let canPlace = !existingCard;
            
            // 如果有组队联手卡牌，可以放置第二个
            if (existingCard && existingCard.ability === 'teamwork') {
                canPlace = true;
            }
            
            // 如果新卡牌有组队联手，也可以放置在有卡牌的位置
            if (existingCard && card.ability === 'teamwork') {
                canPlace = true;
            }
            
            if (canPlace) {
                slot.classList.add('valid-position');
                slot.addEventListener('click', () => this.playCard(card, index));
            }
        });
    }
    
    playCard(card, position) {
        if (this.currentPlayer.mana < card.cost) {
            this.addBattleLog('法力值不足！');
            return;
        }
        
        const existingCard = this.currentPlayer.field[position];
        
        // 检查是否可以放置
        let canPlace = !existingCard;
        
        // 如果有组队联手卡牌，可以放置第二个
        if (existingCard && existingCard.ability === 'teamwork') {
            canPlace = true;
        }
        
        // 如果新卡牌有组队联手，也可以放置在有卡牌的位置
        if (existingCard && card.ability === 'teamwork') {
            canPlace = true;
        }
        
        if (!canPlace) {
            this.addBattleLog('该位置已有卡牌！');
            return;
        }
        
        // 打出卡牌
        this.currentPlayer.field[position] = card;
        card.position = position;
        this.currentPlayer.mana -= card.cost;
        this.currentPlayer.hand.splice(this.selectedCard.index, 1);
        
        this.addBattleLog(`打出了 ${card.name}`);
        this.updateUI();
        
        // 处理抽卡效果
        if (card.ability === 'drawcard' || (card.id === 'sunshroom')) {
            this.currentPlayer.drawCard();
            this.addBattleLog(`${card.name} 发动抽卡效果`);
        }
        
        // 清除选择状态
        this.selectedCard = null;
        document.querySelectorAll('.valid-position').forEach(el => el.classList.remove('valid-position'));
    }
}

// 玩家类
class Player {
    constructor(name, team) {
        this.name = name;
        this.team = team;
        this.health = 20;
        this.mana = 1;
        this.maxMana = 1;
        this.hand = [];
        this.deck = [];
        this.field = new Array(5).fill(null);
    }
    
    initDeck() {
        // 使用默认卡组
        const allCards = [...CARDS.plants, ...CARDS.zombies];
        this.deck = allCards.slice(0, 40); // 简化，取前40张卡
        this.shuffleDeck();
    }
    
    shuffleDeck() {
        for (let i = this.deck.length - 1; i > 0; i--) {
            const j = Math.floor(Math.random() * (i + 1));
            [this.deck[i], this.deck[j]] = [this.deck[j], this.deck[i]];
        }
    }
    
    drawCard() {
        if (this.deck.length === 0) return null;
        
        const card = this.deck.pop();
        const cardInstance = new Card(card);
        this.hand.push(cardInstance);
        return cardInstance;
    }
    
    drawInitialHand() {
        for (let i = 0; i < 3; i++) {
            this.drawCard();
        }
    }
    
    startTurn() {
        this.maxMana = Math.min(10, this.maxMana + 1);
        this.mana = this.maxMana;
        this.drawCard();
        
        // 重置卡牌状态
        this.field.forEach(card => {
            if (card) {
                card.resetTurnState();
            }
        });
    }
    
    playCard(handIndex, position) {
        if (handIndex < 0 || handIndex >= this.hand.length) return false;
        if (position < 0 || position >= 5) return false;
        if (this.field[position]) return false;
        
        const card = this.hand[handIndex];
        if (card.cost > this.mana) return false;
        
        // 打出卡牌
        this.field[position] = card;
        card.position = position;
        this.mana -= card.cost;
        this.hand.splice(handIndex, 1);
        
        return true;
    }
    
    takeDamage(damage) {
        this.health -= damage;
        if (this.health < 0) this.health = 0;
    }
    
    isAlive() {
        return this.health > 0;
    }
    
    getAliveCards() {
        return this.field.filter(card => card && card.isAlive());
    }
    
    removeDeadCards() {
        this.field = this.field.map(card => (card && card.isAlive()) ? card : null);
    }
    
    bounceCardToHand(card) {
        const index = this.field.indexOf(card);
        if (index !== -1) {
            this.field[index] = null;
            card.resetHealth();
            this.hand.push(card);
        }
    }
}

// 卡牌类
class Card {
    constructor(data) {
        Object.assign(this, data);
        this.currentHealth = this.health;
        this.position = -1;
        this.isFrozen = false;
        this.hasAttackedThisTurn = false;
    }
    
    takeDamage(damage, ignoreShield = false) {
        if (!ignoreShield && this.ability === 'shield') {
            damage = Math.max(1, damage - 2);
        }
        
        this.currentHealth -= damage;
        return this.currentHealth <= 0;
    }
    
    isAlive() {
        return this.currentHealth > 0;
    }
    
    canAttack() {
        return this.isAlive() && !this.isFrozen && !this.hasAttackedThisTurn;
    }
    
    resetTurnState() {
        this.isFrozen = false;
        this.hasAttackedThisTurn = false;
    }
    
    freeze() {
        this.isFrozen = true;
    }
    
    resetHealth() {
        this.currentHealth = this.health;
        this.isFrozen = false;
        this.hasAttackedThisTurn = false;
    }
}

// 创建卡牌元素
function createCardElement(card) {
    const cardDiv = document.createElement('div');
    cardDiv.className = `card ${card.team}`;
    
    cardDiv.innerHTML = `
        <div class="card-name">${card.name}</div>
        <div class="card-cost">${card.cost}</div>
        <div class="card-description">${card.description}</div>
        <div class="card-stats">
            <span>⚔️ ${card.attack}</span>
            <span>❤️ ${card.currentHealth}/${card.health}</span>
        </div>
        <div class="card-ability">${ABILITIES[card.ability].name}</div>
    `;
    
    return cardDiv;
}

// 全局游戏状态
let gameState = null;

// 初始化游戏
function initGame() {
    gameState = new GameState();
    gameState.initGame();
    gameState.startTurn();
}

// 结束回合
function endTurn() {
    if (!gameState || !gameState.gameActive) return;
    gameState.endTurn();
}
